@codex please fix

```
Run python desktop-tauri/scripts/test_packaged_operator_e2e.py
Traceback (most recent call last):
  File "/Users/runner/work/token.place/token.place/desktop-tauri/scripts/test_packaged_operator_e2e.py", line 337, in <module>
    raise SystemExit(main())
                     ^^^^^^
  File "/Users/runner/work/token.place/token.place/desktop-tauri/scripts/test_packaged_operator_e2e.py", line 196, in main
    run_model_bridge_inspect_probe(Path(tmpdir))
  File "/Users/runner/work/token.place/token.place/desktop-tauri/scripts/test_packaged_operator_e2e.py", line 93, in run_model_bridge_inspect_probe
    assert result.returncode == 0, combined
           ^^^^^^^^^^^^^^^^^^^^^^
AssertionError: {"ok": false, "error": "Missing Python dependency for model downloads (No module named 'cryptography')."}


Error: Process completed with exit code 1.
```

new codex task, not a r/e/v/i/e/w task

--- a/desktop-tauri/src-tauri/python/desktop_runtime_setup.py
+++ b/desktop-tauri/src-tauri/python/desktop_runtime_setup.py
@@ -7,6 +7,7 @@ import json
 import os
 import subprocess
 import sys
+import tempfile
 import time
 from dataclasses import dataclass
 from pathlib import Path
@@ -606,6 +607,22 @@ def _desktop_dependency_target(runtime_root: Path) -> Path:
     return runtime_root / ".token_place_desktop_site"
 
 
+def _resolve_desktop_dependency_target(runtime_root: Path) -> tuple[Optional[Path], Optional[str]]:
+    preferred_targets = [
+        _desktop_dependency_target(runtime_root),
+        Path.home() / ".token_place_desktop_site",
+        Path(tempfile.gettempdir()) / ".token_place_desktop_site",
+    ]
+    errors: list[str] = []
+    for candidate in preferred_targets:
+        try:
+            candidate.mkdir(parents=True, exist_ok=True)
+            return candidate, None
+        except OSError as exc:
+            errors.append(f"{candidate}: {exc}")
+    return None, "; ".join(errors) if errors else "no writable install target"
+
+
 def ensure_desktop_python_dependencies(*, repo_root: Optional[Path] = None) -> Dict[str, str]:
     """Ensure baseline desktop bridge Python dependencies are importable."""
 
@@ -628,8 +645,17 @@ def ensure_desktop_python_dependencies(*, repo_root: Optional[Path] = None) -> D
         }
 
     env = os.environ.copy()
-    target_dir = _desktop_dependency_target(root)
-    target_dir.mkdir(parents=True, exist_ok=True)
+    target_dir, target_error = _resolve_desktop_dependency_target(root)
+    if target_dir is None:
+        return {
+            "ok": "false",
+            "action": "install_target_unavailable",
+            "missing": ",".join(missing),
+            "interpreter": sys.executable,
+            "import_root": str(root),
+            "requirements": str(requirements_path),
+            "detail": target_error or "unable to create desktop dependency install target",
+        }
     target_dir_str = str(target_dir)
     if target_dir_str not in sys.path:
         sys.path.insert(0, target_dir_str)

--- a/desktop-tauri/src-tauri/python/desktop_runtime_setup.py
+++ b/desktop-tauri/src-tauri/python/desktop_runtime_setup.py
@@ -629,7 +629,7 @@ def ensure_desktop_python_dependencies(*, repo_root: Optional[Path] = None) -> D
     root = _resolve_runtime_root(repo_root=repo_root)
     requirements_path = _resolve_desktop_requirements_path(root)
 
-    required_modules = ("psutil", "requests", "dotenv")
+    required_modules = ("psutil", "requests", "dotenv", "cryptography")
     missing = [name for name in required_modules if importlib.util.find_spec(name) is None]
     if not missing:
         return {"ok": "true", "action": "already_satisfied", "missing": ""}

--- a/desktop-tauri/src-tauri/python/desktop_runtime_setup.py
+++ b/desktop-tauri/src-tauri/python/desktop_runtime_setup.py
@@ -7,7 +7,6 @@ import json
 import os
 import subprocess
 import sys
-import tempfile
 import time
 from dataclasses import dataclass
 from pathlib import Path
@@ -611,7 +610,6 @@ def _resolve_desktop_dependency_target(runtime_root: Path) -> tuple[Optional[Pat
     preferred_targets = [
         _desktop_dependency_target(runtime_root),
         Path.home() / ".token_place_desktop_site",
-        Path(tempfile.gettempdir()) / ".token_place_desktop_site",
     ]
     errors: list[str] = []
     for candidate in preferred_targets:

--- a/tests/unit/test_desktop_runtime_setup.py
+++ b/tests/unit/test_desktop_runtime_setup.py
@@ -855,12 +855,12 @@ def test_ensure_desktop_python_dependencies_reports_requirements_missing(monkeyp
 
     assert result['ok'] == 'false'
     assert result['action'] == 'requirements_missing'
-    assert result['missing'] == 'psutil,requests,dotenv'
+    assert result['missing'] == 'psutil,requests,dotenv,cryptography'
 
 
 def test_ensure_desktop_python_dependencies_reports_install_failed(monkeypatch, tmp_path):
     requirements = tmp_path / 'requirements_desktop_runtime.txt'
-    requirements.write_text('psutil\nrequests\npython-dotenv\n', encoding='utf-8')
+    requirements.write_text('psutil\nrequests\npython-dotenv\ncryptography\n', encoding='utf-8')
     monkeypatch.setattr(desktop_runtime_setup, '_resolve_runtime_root', lambda **_: tmp_path)
     monkeypatch.setattr(desktop_runtime_setup, '_resolve_desktop_requirements_path', lambda _root: requirements)
     monkeypatch.setattr(desktop_runtime_setup.importlib.util, 'find_spec', lambda _name: None)
@@ -884,10 +884,10 @@ def test_ensure_desktop_python_dependencies_reports_install_failed(monkeypatch, 
 
 def test_ensure_desktop_python_dependencies_reports_post_install_missing(monkeypatch, tmp_path):
     requirements = tmp_path / 'requirements_desktop_runtime.txt'
-    requirements.write_text('psutil\nrequests\npython-dotenv\n', encoding='utf-8')
+    requirements.write_text('psutil\nrequests\npython-dotenv\ncryptography\n', encoding='utf-8')
     monkeypatch.setattr(desktop_runtime_setup, '_resolve_runtime_root', lambda **_: tmp_path)
     monkeypatch.setattr(desktop_runtime_setup, '_resolve_desktop_requirements_path', lambda _root: requirements)
-    sequence = iter([None, None, None, object(), object(), None])
+    sequence = iter([None, None, None, None, object(), object(), object(), None])
     monkeypatch.setattr(desktop_runtime_setup.importlib.util, 'find_spec', lambda _name: next(sequence))
     monkeypatch.setattr(desktop_runtime_setup, '_run_pip_install', lambda *_args, **_kwargs: (True, 'ok'))
 
@@ -895,14 +895,14 @@ def test_ensure_desktop_python_dependencies_reports_post_install_missing(monkeyp
 
     assert result['ok'] == 'false'
     assert result['action'] == 'post_install_missing'
-    assert result['missing'] == 'dotenv'
+    assert result['missing'] == 'cryptography'
 
 
 def test_ensure_desktop_python_dependencies_falls_back_to_home_target_when_runtime_root_unwritable(
     monkeypatch, tmp_path
 ):
     requirements = tmp_path / 'requirements_desktop_runtime.txt'
-    requirements.write_text('psutil\nrequests\npython-dotenv\n', encoding='utf-8')
+    requirements.write_text('psutil\nrequests\npython-dotenv\ncryptography\n', encoding='utf-8')
     runtime_root = tmp_path / 'runtime'
     runtime_root.mkdir(parents=True)
     home_dir = tmp_path / 'home'

--- a/tests/unit/test_desktop_runtime_setup.py
+++ b/tests/unit/test_desktop_runtime_setup.py
@@ -896,3 +896,41 @@ def test_ensure_desktop_python_dependencies_reports_post_install_missing(monkeyp
     assert result['ok'] == 'false'
     assert result['action'] == 'post_install_missing'
     assert result['missing'] == 'dotenv'
+
+
+def test_ensure_desktop_python_dependencies_falls_back_to_home_target_when_runtime_root_unwritable(
+    monkeypatch, tmp_path
+):
+    requirements = tmp_path / 'requirements_desktop_runtime.txt'
+    requirements.write_text('psutil\nrequests\npython-dotenv\n', encoding='utf-8')
+    runtime_root = tmp_path / 'runtime'
+    runtime_root.mkdir(parents=True)
+    home_dir = tmp_path / 'home'
+    home_dir.mkdir(parents=True)
+
+    monkeypatch.setattr(desktop_runtime_setup, '_resolve_runtime_root', lambda **_: runtime_root)
+    monkeypatch.setattr(desktop_runtime_setup, '_resolve_desktop_requirements_path', lambda _root: requirements)
+    monkeypatch.setattr(desktop_runtime_setup.importlib.util, 'find_spec', lambda _name: None)
+    monkeypatch.setattr(desktop_runtime_setup.Path, 'home', staticmethod(lambda: home_dir))
+    captured = {}
+
+    def _capture_run(cmd, *_args, **_kwargs):
+        captured['cmd'] = cmd
+        return False, 'install failed: boom'
+
+    original_mkdir = desktop_runtime_setup.Path.mkdir
+
+    def _fake_mkdir(path_obj, parents=False, exist_ok=False):
+        if str(path_obj).endswith('.token_place_desktop_site') and runtime_root in path_obj.parents:
+            raise PermissionError('read-only bundle')
+        return original_mkdir(path_obj, parents=parents, exist_ok=exist_ok)
+
+    monkeypatch.setattr(desktop_runtime_setup.Path, 'mkdir', _fake_mkdir)
+    monkeypatch.setattr(desktop_runtime_setup, '_run_pip_install', _capture_run)
+
+    result = desktop_runtime_setup.ensure_desktop_python_dependencies(repo_root=runtime_root)
+
+    assert result['action'] == 'install_failed'
+    assert '--target' in captured['cmd']
+    target_idx = captured['cmd'].index('--target') + 1
+    assert captured['cmd'][target_idx] == str(home_dir / '.token_place_desktop_site')

